vortex-array/src/array/struct_/compute.rs

```diff
@@ -82,7 +82,10 @@ impl FilterFn for StructArray {
             .children()
             .map(|field| filter(&field, predicate))
             .try_collect()?;
-        let length = fields.first().map(|a| a.len()).unwrap_or_default();
+        let length = fields
+            .first()
+            .map(|a| a.len())
+            .ok_or(vortex_err!("Struct arrays should have at least one field"))?;
```

Comment:
Ok_or_else. There really should be a clippy rule for this

Reply:
of course you're right :facepalm: